### ISSUE TITLE
Create template repo / cookiecutter template

### ISSUE THREAD

**Issue by trey-stafford** [QGreenland-Net/.github#22]
Create a template repo / cookiecutter template with the following elements:

* licence
* code of conduct
* README w/
    * Mermaid diagram at the top, followed immediately by a link to qgreenland-net.github.io to see system-level diagram/docs
    * attribution for ADC, NSDIC
* ...

Cookiecutter project might be good for having multiple repo types (e.g., `python` and `r`). For example, I'd like to be able to quickly create a repo w/ all the boilerplate I like for python (e.g., an `environment.yml` with common deps, `pre-commit` config to run `ruff`, etc.).

**Comment by mfisher87** [comment]
@mbjones do you have any prior work that we can leverage? We have a GitHub template repository at NSIDC, but it's private. It's pretty basic, only includes a README and LICENSE templates.

**Comment by mbjones** [comment]
No, we don't have a common template as we collaborate with some many groups with different needs. In general, we opt for Apache2 licensing all things being equal. Here are two repos with a common layout from our work that could be made into a template: 

- JS app: https://github.com/NCEAS/metacatui
- Python app: https://github.com/DataONEorg/hashstore/tree/main

In particular, these both have detailed CONTRIBUTING.md documents that document our typical development workflow.

**Comment by rmarow** [comment]
Template linked here: https://github.com/QGreenland-Net/template 
Note: decided it was probably best to create a non-language specific template to start. Want to encourage the use of any necessary languages.

**Comment by rmarow** [comment]
Closing for now will revisit some licensing with Matt jones at a later date.
